tests/parquet_test.py

```diff
@@ -534,6 +534,18 @@ def test_segarray_string(self):
             rd_data = ak.read_parquet(f"{tmp_dirname}/segarray_test_empty_*")
             self.assertListEqual(s.to_list(), rd_data.to_list())
 
+    def test_float_edge(self):
+        df = pd.DataFrame({"FloatList": [[3.14, np.nan, 2.23], [], [3.08], [np.inf, 6.8], [-0.0]]})
```

Comment:
I pulled this down to mess around a bit with the nan in different spots and i'm seeing `0.0` instead of `nan` for all nan after the last non-`nan` is read
```python
df = pd.DataFrame({"FloatList": [[3.14, 2.23], [], [3.08], [np.inf, 6.8], [-0.0, np.nan, np.nan, np.nan]]})
```

printing out `pd_l` and `ak_l` shows the difference
```
pd_l: [[3.14, 2.23], [], [3.08], [inf, 6.8], [-0.0, nan, nan, nan]]
ak_l: [[3.14, 2.23], [], [3.08], [inf, 6.8], [-0.0, 0.0, 0.0, 0.0]]
```